apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/ApixV1Webscript.java

```diff
@@ -81,6 +82,24 @@ protected NodeInfo nodeRefToNodeInfo(NodeRef nodeRef, IFileFolderService fileFol
         return nodeInfoList;
     }
 
+    protected NodeInfo nodeRefToNodeInfo(NodeRef nodeRef,
```

Comment:
If you have 2 conversion methods --one for a single object and one for lists-- then the list one should use the single one, not vice versa IMO.

Reply:
The way I see it a single object is a special case of multiple objects. I know that in some cases following this philosophy can be more efficient. For example if you have a method that handles one object and does some network calls. If you have to do this for multiple objects it would mean you have to do those calls also multiple times because you loop over the same method multiple times. So the more objects you have the more network calls are needed. Instead, if you would handle the single object as if it were multiple objects then you could in theory always do this in the same amount of calls. This is just an example but I think it illustrates the difference.

Do you have other reasons why a method for multiple objects should use the one for a single object?

Reply:
From an **API perspective** I agree with you, and it is a good principle to follow. But this is not an API, but internal code structure (so network overhead is not relevant). 

The usual way of writing code is *loop conversion > foreach of single conversion*. This is *single conversion > transform to list -> loop conversion -> transform back to single*
Not only does that deviate from what most readers would expect, it introduces unnecessary complexity of 2 conversions.

Reply:
@WimCrols I think it depends on the point of view and approach. I think there are 2 options:
1. Collection conversion is foreach of single conversion
2. Single conversion is a special case of collection conversion
Both are valid options and they have both pros and cons. Depending on use case, complexity of processing a single conversion and  performance overhead/considerations one of the options is preferable.

I would follow the general convention used in this project for implementing methods covering both single and collection arguments.